internal/docker/deployment.go

```diff
@@ -57,6 +58,16 @@ func (hsDep *HomeserverDeployment) SetEndpoints(baseURL string, fedBaseURL strin
 	}
 }
 
+func (d *Deployment) GetFullyQualifiedHomeserverName(t ct.TestLike, hsName string) spec.ServerName {
+	_, ok := d.HS[hsName]
+	if !ok {
+		ct.Fatalf(t, "Deployment.GetFullyQualifiedHomeserverName - HS name '%s' not found", hsName)
+	}
+	// We have network aliases for each Docker container that will resolve the `hsName` to
+	// the container.
+	return spec.ServerName(hsName)
+}
+
```

Comment:
Currently, I don't think the built-in Complement `Deployment` implementation supports multiple `Deployments` at the same time (`hs1`, `hs2` would conflict between them). Since one of the goals of this PR is to unlock that functionality for custom `Deployment`'s, it could make some sense to also refactor and update that here as well.

I'd rather leave it as-is until we need it or at-least do this in a follow-up PR.

See the PR description for more context on multiple `Deployment`.

Reply:
The PR description doesn't give more context on why multiple `Deployment` is desirable:

> But imagine a case where we have multiple Deployment and we want the homeservers to communicate with each other. 

_Why_ would you do this? The `Deployment` is meant to encapsulate an _entire_ deployment, all servers and network links between them. I don't understand the rationale.